Xamarin.Essentials/Calendar/Calendar.ios.cs

```diff
@@ -0,0 +1,127 @@
+            var sDate = startDateToConvert.ToNSDate();
+            var eDate = endDateToConvert.ToNSDate();
+            EKCalendar[] calendars;
+            try
+            {
+                calendars = !string.IsNullOrWhiteSpace(calendarId)
+                    ? CalendarRequest.Instance.Calendars.Where(x => x.CalendarIdentifier == calendarId).ToArray()
+                    : null;
+            }
+            catch (NullReferenceException ex)
+            {
+                throw new NullReferenceException($"iOS: Unexpected null reference exception {ex.Message}");
+            }
+
+            var query = CalendarRequest.Instance.PredicateForEvents(sDate, eDate, calendars);
+            var events = CalendarRequest.Instance.EventsMatching(query);
+
+            foreach (var e in events)
+            {
+                eventList.Add(new DeviceEvent
+                {
+                    Id = e.CalendarItemIdentifier,
+                    CalendarId = e.Calendar.CalendarIdentifier,
+                    Title = e.Title,
+                    StartDate = e.StartDate.ToDateTimeOffset(),
+                    EndDate = !e.AllDay ? (DateTimeOffset?)e.EndDate.ToDateTimeOffset() : null
+                });
+            }
+            eventList.Sort((x, y) =>
+            {
+                if (!y.EndDate.HasValue)
+                {
+                    return 0;
+                }
+                return x.StartDate.CompareTo(y.EndDate.Value);
+            });
+
+            return eventList;
+        }
+
+        static async Task<DeviceEvent> PlatformGetEventByIdAsync(string eventId)
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            EKEvent e;
+            try
+            {
+                e = CalendarRequest.Instance.GetCalendarItem(eventId) as EKEvent;
+            }
+            catch (NullReferenceException)
+            {
+                throw new NullReferenceException($"[iOS]: No Event found for event Id {eventId}");
+            }
+
+            return new DeviceEvent
+            {
+                Id = e.CalendarItemIdentifier,
+                CalendarId = e.Calendar.CalendarIdentifier,
+                Title = e.Title,
+                Description = e.Notes,
+                Location = e.Location,
+                StartDate = e.StartDate.ToDateTimeOffset(),
+                EndDate = !e.AllDay ? (DateTimeOffset?)e.EndDate.ToDateTimeOffset() : null,
+                Attendees = e.Attendees != null ? GetAttendeesForEvent(e.Attendees) : new List<DeviceEventAttendee>()
+            };
+        }
+
+        static IEnumerable<DeviceEventAttendee> GetAttendeesForEvent(IEnumerable<EKParticipant> inviteList)
+        {
+            var attendees = new List<DeviceEventAttendee>();
+
+            foreach (var attendee in inviteList)
```

Comment:
linq

Reply:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43

Added sort by name aswell